pypesto/profile/profile.py

```diff
@@ -86,139 +89,64 @@ def create_next_guess(x, par_index, par_direction_, profile_options_,
     # existing list of profiles
     global_opt = initialize_profile(problem, result, result_index,
                                     profile_index, profile_list)
-
-    # loop over parameters for profiling
-    for i_par in profile_index:
-        # only compute profiles for free parameters
-        if i_par in problem.x_fixed_indices:
-            continue
-
-        # create an instance of ProfilerResult, which will be appended to the
-        # result object, when this profile is finished
-        current_profile = result.profile_result.get_profiler_result(
-            i_par=i_par, profile_list=profile_list)
-
-        # compute profile in descending and ascending direction
-        for par_direction in [-1, 1]:
-            # flip profile
-            current_profile.flip_profile()
-
-            # compute the current profile
-            current_profile = walk_along_profile(
+    # if engine==None do the original profiling
+    if engine is None:
```

Comment:
would it be possible to define a "dummy engine" which just executes the for loop sequentially? it could use the same ProfilerTask definition as below, so one would simplify the code by not having that flip_profile, walk_along_profile etc twice.

Reply:
could just be SingleCoreEngine, right?

Reply:
which should then be the default.

Reply:
Could do, but normally the engine used would be the `SingleCoreEngine` wouldn't it? I thought that if it get approved, i would delete the whole block and put instead:
```python
if engine is None: engine = SingleCoreEngine()
```
That is at least the case in the optimize function.

Reply:
oh ok, the messages did not load. Will do so 👍

Reply:
yes, exactly. please go ahead and remove the old code already, then a review will be easier.

Reply:
@yannikschaelte should be done now.